src/main/scala/tech/cryptonomic/conseil/Lorre.scala

```diff
@@ -450,6 +462,107 @@ object Lorre extends App with TezosErrors with LazyLogging with LorreAppConfig w
     }
   }
 
+  /**
+    * Collect data from Blocks Table, Operations Table, and the Tezos RPC,
+    * return the appropriate list of VoteAggregates, and write them to the Votes table.
+    * @param ex the needed ExecutionContext to combine multiple database operations
+    * @return a future result of the number of rows stored to db, if supported by the driver
+    */
+  private[this] def processTezosVotes(): Future[Done] = {
+    import slick.jdbc.PostgresProfile.api._
+    implicit val mat = ActorMaterializer()
+
+    logger.info("Processing latest Tezos votes data...")
+
+    /**
+      * * Helper function, takes a sequence of GovernanceData, separates them by ballot types.
+      * @param votingData
+      * @return
+      */
+    def partitionByVote(votingData: Seq[VotingData]): (Seq[VotingData], Seq[VotingData], Seq[VotingData]) = {
+      val (yays, notYays) = votingData.partition(x => x.ballot.contains("yay"))
+      val (nays, notYaysOrNays) = notYays.partition(x => x.ballot.contains("nay"))
+      val (passes, _) = notYaysOrNays.partition(x => x.ballot.contains("pass"))
+      (yays, nays, passes)
+    }
+
+    /**
+      * * Helper function, takes a sequence of GovernanceData, calculates stake.
+      * @param votingData
+      * @return
+      */
+    def calculateStake(votingData: Seq[VotingData]): Future[BigDecimal] =
+      Future
+        .sequence(
+          votingData.map {
+            case VotingData(_, _, hash, _, _, _, address) =>
+              tezosNodeOperator.getAccountBalanceForBlock(BlockHash(hash), AccountId(address.get))
+          }
+        )
+        .map(balances => balances.reduceOption(_ + _).getOrElse(0))
+
+    /**
+      * * Helper function, takes a sequence of GovernanceData, processes aggregate fields.
+      * @param votingData
+      * @return
+      */
+    def processVoteAggregate(votingData: Seq[VotingData]): Future[TezosTypes.VoteAggregates] = {
+      val sampleVotingField = votingData.head
+      val timestamp = sampleVotingField.timestamp
+      val cycle = sampleVotingField.cycle
+      val level = sampleVotingField.level
+      val proposalHash = sampleVotingField.proposalHash
+      val (yays, nays, passes) = partitionByVote(votingData)
+      val yayCount = yays.length
+      val nayCount = nays.length
+      val passCount = passes.length
+      for {
```

Comment:
Here I guess we can profit from starting the Futures immediately and not within the for comprehension, to parallelize work and get results a bit faster

Reply:
ok, changed